### DISCUSSION TITLE
请问如何自定义UP角色以及武器，或者调整概率呢

### DISCUSSION THREAD

**Discussion by ID114514** [Arondight/Adachi-BOT#219] [Q&A] [answered]
想要通过这种方式复刻旧版卡池

**Comment by Arondight** [comment, marked as answer]
建议放弃。

首先你要禁止卡池更新。

https://github.com/Arondight/Adachi-BOT/blob/d3a78c85befb66317cc6ebe7040626f4857bab36/src/utils/init.js#L51

https://github.com/Arondight/Adachi-BOT/blob/d3a78c85befb66317cc6ebe7040626f4857bab36/src/utils/init.js#L54

然后更改 `./data/db/gacha.json` 中卡池的字段，更改完毕后重启 bot 。

**Reply by Arondight** [reply]
最简单的办法如下。

```patch
diff --git a/src/utils/update.js b/src/utils/update.js
index 174c817..344c27d 100644
--- a/src/utils/update.js
+++ b/src/utils/update.js
@@ -42,6 +42,8 @@ async function parseData(gachaID) {
 }

 async function gachaUpdate() {
+  return;
+
   const gachaInfo = (await getGachaList()).data.list;

   if (undefined === gachaInfo[1]) {
```

**Comment by Arondight** [comment]
在下面的代码中修改出货概率。

https://github.com/Arondight/Adachi-BOT/blob/d3a78c85befb66317cc6ebe7040626f4857bab36/src/plugins/gacha/gacha.js#L43-L71
